### PR TITLE
Allowance Transfer Fixes

### PR DESCRIPTION
**Pull request by mhess-swl**
This PR fixes some errors in `CryptoTransferSuite.allowanceTransfersWorkAsExpected()`. Due to an issue with determining repeated accounts, however, this test can't be enabled yet.

### CODE REVIEW

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java [hunk 1]
```diff
@@ -887,7 +887,13 @@ private HapiSpec allowanceTransfersWorkAsExpected() {
                                 .payingWith(SPENDER)
                                 .signedBy(SPENDER)
                                 .fee(ONE_HBAR)
-                                .hasKnownStatus(INSUFFICIENT_SENDER_ACCOUNT_BALANCE_FOR_CUSTOM_FEE),
+                                // INSUFFICIENT_SENDER_ACCOUNT_BALANCE_FOR_CUSTOM_FEE is the expected mono service
+                                // outcome here, but the modularized implementation doesn't provide enough information
```
Comment (Neeharika-Sompalli) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java:891]: Are we not enabling any tests now ?

Comment (mhess-swl) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java:891]: No, it's blocked by the repeated accounts issue

Comment (Neeharika-Sompalli) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java:891]: Ah okayy! 😞

Comment (mhess-swl) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java:891]: Repeated accounts issue: #9321

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/AssociateTokenRecipientsStep.java [hunk 2]
```diff
@@ -88,10 +91,34 @@ public void doIn(@NonNull final TransferContext transferContext) {
...
+                    }
+                }
+
+                // If no allowance or spender is found, check the NFT's owner
+                if (!allowanceFound) {
+                    if (nft.hasOwnerId()) {
+                        validateTrue(nft.ownerIdOrThrow().equals(senderId), SENDER_DOES_NOT_OWN_NFT_SERIAL_NO);
+                    } else {
+                        // If there's no owner, this NFT's only authorized sender is the token treasury
+                        validateTrue(token.treasuryAccountId().equals(senderId), SENDER_DOES_NOT_OWN_NFT_SERIAL_NO);
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/AssociateTokenRecipientsStep.java:120]: I think this logic is already there in `NFTOwnersChangeStep`. Not sure if we need these here

Comment (mhess-swl) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/AssociateTokenRecipientsStep.java]: I refactored the code so the same check is being done in both places, without duplicating the logic. Good call! 👍

**Comment by mhess-swl** [comment]
Closing in favor of #9334

**Pull request closed without merging**